lib/ansible/modules/system/portenforce.py

```diff
@@ -0,0 +1,196 @@
+                    results.append(result)
+            elif not pids:
+                raise EnvironmentError('Could not get the pids for the listening ports - possibly a permission issue')
+    return results
+
+def applyWhitelist(portspids, whitelist=list()):
+    kill_pids = list()
+    for p in portspids:
+        if int(p['port']) not in whitelist and str(p['port']) not in whitelist:
+            to_kill = dict(pid=p['pid'], port=p['port'], proto=p['proto'], name=p['name'], stime=p['stime'])
+            if to_kill not in kill_pids:
+                kill_pids.append(to_kill)
+    return kill_pids
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            whitelist_tcp = dict(required=False, type='list', default=list()),
+            whitelist_udp = dict(required=False, type='list', default=list())
+        ),
+        supports_check_mode=True
+    )
+
+    if platform.system() != 'Linux':
+        module.fail_json(msg='This module requires Linux.')
+
+    if not module.params['whitelist_tcp'] and not module.params['whitelist_udp']:
+        module.fail_json(msg="You must supply either a 'whitelist_tcp' or 'whitelist_udp' argument.")
+
+    def getPidSTime(pid):
+        ps_cmd = module.get_bin_path('ps', True)
+        p1 = Popen([ps_cmd, '-o', 'lstart', '-p', str(pid)], stdout=PIPE, stderr=PIPE)
+        ps_output = p1.communicate()[0]
+        stime = ''
+        for line in iter(ps_output.splitlines()):
+            if 'started' not in line:
+                stime = line
+        return stime
+
+    result = {}
+    result['changed'] = False
+    result['killed'] = list()
+
+    try:
+        netstat_cmd = module.get_bin_path('netstat', True)
+
+        # which TCP ports are listening for connections?
+        p1 = Popen([netstat_cmd, '-plnt'], stdout=PIPE, stderr=PIPE)
+        output_tcp = p1.communicate()[0]
+        kill_tcp = netStatParse(output_tcp)
+        for i, p in enumerate(kill_tcp):
+            p['stime'] = getPidSTime(p['pid']) 
+            kill_tcp[i] = p
+
+        # which UDP ports are listening for connections?
+        p1 = Popen([netstat_cmd, '-plnu'], stdout=PIPE, stderr=PIPE)
+        output_udp = p1.communicate()[0]
+        kill_udp = netStatParse(output_udp)
+        for i, p in enumerate(kill_udp):
+            p['stime'] = getPidSTime(p['pid']) 
+            kill_udp[i] = p
+
+    except EnvironmentError:
+        err = get_exception()
+        module.fail_json(msg=str(err))
+
+    # gather details of the pids to kill
+    kill_pids_tcp = applyWhitelist(kill_tcp, module.params['whitelist_tcp'])
+    kill_pids_udp = applyWhitelist(kill_udp, module.params['whitelist_udp'])
+    kill_all = list(kill_pids_tcp)
+    kill_all.extend(x for x in kill_pids_udp if x not in kill_pids_tcp)
+
+    # get just the pids to avoid duplicates across protocols
+    kill_unique = list()
+    for p in kill_all:
+        if p['pid'] not in kill_unique:
+            kill_unique.append(p)
```

Comment:
Good pick up, I can't see how it would work! Originally the `kill_unique` list was only storing pids but it turned into dicts when the need to check the pid start time was identified, and this unique check wasn't adjusted.